File: src/main/java/com/freya02/botcommands/parameters/ParameterResolver.java
package com.freya02.botcommands.parameters;

import com.freya02.botcommands.Logging;
import org.slf4j.Logger;

/**
 * Base class for parameter resolvers used in regex commands, slash commands and buttons callbacks
 *
 * @see RegexParameterResolver
 * @see SlashParameterResolver
 * @see ComponentParameterResolver
 */
public abstract class ParameterResolver {
	protected final Logger LOGGER = Logging.getLogger(this);

	private final Class<?> type;

	public ParameterResolver(Class<?> type) {
		this.type = type;
	}

	public Class<?> getType() {
		return type;
	}
}

File: src/main/java/com/freya02/botcommands/api/parameters/ParameterResolver.kt
package com.freya02.botcommands.api.parameters

import com.freya02.botcommands.api.Logging
import org.slf4j.Logger
import kotlin.reflect.KType

/**
 * Base class for parameter resolvers used in regex commands, application commands and buttons callbacks
 *
 * @see RegexParameterResolver
 * @see QuotableRegexParameterResolver
 * @see ComponentParameterResolver
 * @see SlashParameterResolver
 * @see MessageContextParameterResolver
 * @see UserContextParameterResolver
 */
abstract class ParameterResolver {
    @JvmField
    protected val LOGGER: Logger

    val type: KType

    /**
     * Constructs a new parameter resolver
     *
     * @param type Type of the parameter being resolved
     */
    constructor(type: ParameterType) {
        this.type = type.type
        this.LOGGER = Logging.getLogger(this)
    }
}